### PR TITLE
Python 3.{8,9} porting round 2

### PR DESCRIPTION
**Pull request by jsmolic**

### CODE REVIEW

File: app-misc/gcalcli/gcalcli-4.2.0.ebuild [hunk 1]
```diff
@@ -1,33 +1,30 @@
...
 
+PYTHON_COMPAT=( python3_{7..9} )
+DISTUTILS_USE_SETUPTOOLS=rdepend
 inherit distutils-r1
 
 MY_PV="4.2.0"
-
 DESCRIPTION="Google Calendar Command Line Interface"
 HOMEPAGE="https://github.com/insanum/gcalcli"
+S="${WORKDIR}/gcalcli-${MY_PV}"
```
Comment (SoapGentoo) [inline app-misc/gcalcli/gcalcli-4.2.0.ebuild:13]: comes after `SRC_URI`

File: app-misc/gcalcli/metadata.xml [hunk 2]
```diff
@@ -5,7 +5,9 @@
     <email>robbat2@gentoo.org</email>
     <name>Robin H. Johnson</name>
   </maintainer>
+  </stabilize-allarches>
   <upstream>
     <remote-id type="github">insanum/gcalcli</remote-id>
+	<remote-id type="pypi">gcalcli</remote-id>
```
Comment (SoapGentoo) [inline app-misc/gcalcli/metadata.xml:11]: is this correctly formatted?

File: dev-python/ansi/ansi-0.2.0.ebuild [hunk 3]
```diff
@@ -1,20 +1,19 @@
...
 
-PYTHON_COMPAT=( python{3_7,3_8} )
-
+PYTHON_COMPAT=( python3_{7..9} )
+DISTUTILS_USE_SETUPTOOLS=no
 inherit distutils-r1
 
 DESCRIPTION="ANSI cursor movement and graphics in Python"
 HOMEPAGE="https://github.com/tehmaze/ansi"
+S="${WORKDIR}/${PN}-${P}"
```
Comment (SoapGentoo) [inline dev-python/ansi/ansi-0.2.0.ebuild:12]: ditto

File: app-misc/gcalcli/metadata.xml (hunk 2 above)
Comment (thesamesam) [inline app-misc/gcalcli/metadata.xml:11]: Indentation

File: dev-python/ansi/ansi-0.2.0.ebuild [hunk 4]
```diff
@@ -1,20 +1,19 @@
...
 SRC_URI="https://github.com/tehmaze/${PN}/archive/${P}.tar.gz"
```
Comment (thesamesam) [inline dev-python/ansi/ansi-0.2.0.ebuild:13]: SRC_URI then S

File: dev-python/nbval/nbval-0.9.6.ebuild [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+DESCRIPTION="A py.test plugin to validate Jupyter notebooks"
+HOMEPAGE="https://github.com/computationalmodelling/nbval"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="
+	dev-python/pytest[${PYTHON_USEDEP}]
```
Comment (thesamesam) [inline dev-python/nbval/nbval-0.9.6.ebuild:18]: No need! Also, sort.

File: dev-python/rpy/rpy-3.4.2.ebuild [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+inherit distutils-r1 flag-o-matic virtualx
+
+MYSLOT=2
+MY_PN=${PN}${MYSLOT}
+MY_P=${MY_PN}-${PV}
+
+DESCRIPTION="Python interface to the R Programming Language"
+HOMEPAGE="https://rpy.sourceforge.net/
+	https://pypi.org/project/rpy2/"
+S="${WORKDIR}/${MY_P}"
```
Comment (thesamesam) [inline dev-python/rpy/rpy-3.4.2.ebuild:17]: SRC_URI then S (apply to all packages)

**Review by thesamesam — COMMENTED** [review]
I couldn't help myself (off for a rest now) but I saw the email pop up!

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: dev-python/nbval/nbval-0.9.6.ebuild [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+	dev-python/six[${PYTHON_USEDEP}]
+	dev-python/jupyter_client[${PYTHON_USEDEP}]
+	dev-python/nbformat[${PYTHON_USEDEP}]
+	dev-python/ipykernel[${PYTHON_USEDEP}]
+	dev-python/coverage[${PYTHON_USEDEP}]"
```
Comment (thesamesam) [inline dev-python/nbval/nbval-0.9.6.ebuild:23]: Do we definitely need coverage at runtime, actually?

File: dev-python/pymssql/pymssql-2.1.5.ebuild [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+DESCRIPTION="Simple MSSQL python extension module"
+HOMEPAGE="https://www.pymssql.org/ https://pypi.org/project/pymssql/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="LGPL-2.1"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~hppa ~ia64 ~ppc ~ppc64 ~s390 ~sparc ~x86"
+IUSE="test"
+
+# tests need a running instance of freetds
```
Comment (thesamesam) [inline dev-python/pymssql/pymssql-2.1.5.ebuild:18]: Not that I care here, but you can do something like in the hiredis ebuild to try spawn a server (just for your reference)

File: net-im/err/err-6.1.7.ebuild [hunk 9]
```diff
@@ -45,35 +46,38 @@ RDEPEND="
...
 
 	distutils-r1_python_prepare_all
 }
 
+python_test() {
+	local deselect=(
+		--deselect tests/commands_test.py::test_plugin_cycle
+		--deselect tests/commands_test.py::test_broken_plugin
+		--deselect tests/commands_test.py::test_backup
+		--deselect tests/plugin_management_test.py::test_check_dependencies_requi
```
Comment (thesamesam) [inline net-im/err/err-6.1.7.ebuild:74]: Weird test name...

File: net-irc/limnoria/limnoria-20200124.ebuild [hunk 10]
```diff
@@ -1,34 +1,36 @@
...
-	SRC_URI=""
 else
+	S="${WORKDIR}/${MY_PN}-master-${MY_PV}"
 	SRC_URI="https://github.com/ProgVal/${MY_PN}/archive/master-${MY_PV}.tar.gz -> ${P}.tar.gz"
 	KEYWORDS="~amd64 ~x86"
-	S="${WORKDIR}/${MY_PN}-master-${MY_PV}"
 fi
 
 DESCRIPTION="Python based extensible IRC infobot and channel bot"
 HOMEPAGE="https://supybot.aperio.fr/"
```
Comment (thesamesam) [inline net-irc/limnoria/limnoria-20200124.ebuild:25]: Not their HOMEPAGE anymore (and Limnoria is a fork of supybot)

File: dev-python/nbval/nbval-0.9.6.ebuild (hunk 5 above)
Comment (jsmolic) [reply dev-python/nbval/nbval-0.9.6.ebuild:18]: I think this one needs pytest at runtime (https://github.com/computationalmodelling/nbval/blob/master/setup.py#L24)

File: dev-python/nbval/nbval-0.9.6.ebuild (hunk 7 above)
Comment (jsmolic) [reply dev-python/nbval/nbval-0.9.6.ebuild:23]: https://github.com/computationalmodelling/nbval/blob/master/setup.py#L28

File: dev-python/nbval/nbval-0.9.6.ebuild (hunk 5 above)
Comment (thesamesam) [reply dev-python/nbval/nbval-0.9.6.ebuild]: Oops. Ignore me.

File: app-misc/gcalcli/metadata.xml (hunk 2 above)
Comment (jsmolic) [reply app-misc/gcalcli/metadata.xml]: done

File: dev-python/nbval/nbval-0.9.6.ebuild (hunk 7 above)
Comment (thesamesam) [reply dev-python/nbval/nbval-0.9.6.ebuild]: Fine 😞

File: app-misc/gcalcli/metadata.xml (hunk 2 above)
Comment (thesamesam) [reply app-misc/gcalcli/metadata.xml]: _squints_

File: dev-python/ansi/ansi-0.2.0.ebuild (hunk 4 above)
Comment (jsmolic) [reply dev-python/ansi/ansi-0.2.0.ebuild:12]: done

File: app-misc/gcalcli/metadata.xml (hunk 2 above)
Comment (jsmolic) [reply app-misc/gcalcli/metadata.xml]: it is now :smile:

File: dev-python/ansi/ansi-0.2.0.ebuild (hunk 3 above)
Comment (jsmolic) [reply dev-python/ansi/ansi-0.2.0.ebuild]: done

File: net-irc/limnoria/limnoria-20200124.ebuild (hunk 10 above)
Comment (jsmolic) [reply net-irc/limnoria/limnoria-20200124.ebuild]: Thanks, I wasn't checking

Comment (thesamesam) [reply net-irc/limnoria/limnoria-20200124.ebuild]: I'm an IRC obsessive which is why I know :)

File: net-im/err/err-6.1.7.ebuild (hunk 9 above)
Comment (jsmolic) [reply net-im/err/err-6.1.7.ebuild:74]: yeah..

**Comment by mm1ke** [comment]
Some thoughts: You might want to look at some of the packages license too. 
For example `net-misc/pyhoca-gui` has the license `AGPL-3` while the sources indicate it should be `AGPL-3+` (the source files I've checked all contain `published by the Free Software Foundation; either version 3 of the License, or (at your option) any later version.`)

**Comment by thesamesam** [comment]
Great work, thank you!

**Pull request closed without merging**